wa/instruments/perf.py

```diff
@@ -31,51 +32,73 @@ class PerfInstrument(Instrument):
+    on android devices since it recognises android symbols in record mode and is much more stable
+    when reporting record .data files. For more information see simpleperf documentation at:
+    https://android.googlesource.com/platform/system/extras/+/master/simpleperf/doc/README.md
 
     Performance counters are CPU hardware registers that count hardware events
     such as instructions executed, cache-misses suffered, or branches
     mispredicted. They form a basis for profiling applications to trace dynamic
     control flow and identify hotspots.
 
-    pref accepts options and events. If no option is given the default '-a' is
-    used. For events, the default events are migrations and cs. They both can
-    be specified in the config file.
+    perf accepts options and events. If no option is given the default '-a' is
+    used. For events, the default events for perf are migrations and cs. The default
+    events for simpleperf are raw-cpu-cycles, raw-l1-dcache, raw-l1-dcache-refill, raw-instructions-retired.
+    They both can be specified in the config file.
 
     Events must be provided as a list that contains them and they will look like
     this ::
 
-        perf_events = ['migrations', 'cs']
+        (for perf_type = perf ) perf_events = ['migrations', 'cs']
+        (for perf_type = simpleperf) perf_events = ['raw-cpu-cycles', 'raw-l1-dcache']
+
 
     Events can be obtained by typing the following in the command line on the
     device ::
 
         perf list
+        simpleperf list
 
     Whereas options, they can be provided as a single string as following ::
 
         perf_options = '-a -i'
+        perf_options = '--app com.adobe.reader'
 
     Options can be obtained by running the following in the command line ::
 
         man perf-stat
     """
 
     parameters = [
-        Parameter('events', kind=list_of_strs, default=['migrations', 'cs'],
-                  global_alias='perf_events',
-                  constraint=(lambda x: x, 'must not be empty.'),
+        Parameter('perf_type', kind=str, allowed_values=['perf', 'simpleperf'], default='perf',
+                  global_alias='perf_type', description="""Specifies which type of perf binaries
+                  to install. Use simpleperf for collecting perf data on android systems."""),
+        Parameter('command', kind=str, default='stat', allowed_values=['stat', 'record'],
+                  global_alias='perf_command', description="""Specifies which perf command to use. If in record mode
+                  report command will also be executed and results pulled from target along with raw data
+                  file"""),
+        Parameter('events', kind=list_of_strs, global_alias='perf_events',
                   description="""Specifies the events to be counted."""),
-        Parameter('optionstring', kind=list_or_string, default='-a',
+        Parameter('option_string', kind=list_or_string, default='-a',
```

Comment:
Renaming existing parameter will a compatibility breaking change, please undo.

Reply:
Ah sorry, reverted back to original.